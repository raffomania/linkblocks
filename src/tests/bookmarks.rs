use sqlx::{Pool, Postgres};

use crate::tests::util::test_app::TestApp;

#[test_log::test(sqlx::test)]
async fn get_unsorted_bookmarks(pool: Pool<Postgres>) -> anyhow::Result<()> {
  let mut app = TestApp::new(pool).await;
  app.create_test_user().await;
  app.login_test_user().await;

  let unsorted_bookmarks = app.req().get("/bookmarks/unsorted").await.test_page().await;

  insta::assert_snapshot!(unsorted_bookmarks.dom.htmls());

  Ok(())
}
