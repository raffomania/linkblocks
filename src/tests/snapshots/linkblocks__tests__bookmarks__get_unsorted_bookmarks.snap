---
source: src/tests/bookmarks.rs
expression: unsorted_bookmarks.dom.htmls()
---
  <head>
    <link rel="stylesheet" href="/assets/preflight.css" />
    <link rel="stylesheet" href="/assets/railwind.css" />
    <script src="/assets/htmx.1.9.9.js"></script>
    <meta name="color-scheme" content="dark" />
    <meta name="viewport" content="width=device-width,initial-scale=1" />
    <title>linkblocks</title>
  </head>
  <body class="w-full h-full text-gray-200 bg-neutral-800">
    <div class="flex-row-reverse h-full sm:flex">
      <main class="sm:overflow-y-auto sm:grow">
        <header class="px-4 pt-3 mb-4">
          <h1 class="text-xl font-bold">Unsorted Bookmarks</h1>
        </header>
      </main>
      <aside id="nav" class="bg-neutral-900 sm:max-w-[18rem] sm:w-1/3 sm:max-h-full flex flex-col sm:flex-col-reverse sm:border-r border-neutral-700 border-t sm:border-t-0">
        <div class="sm:overflow-y-auto sm:flex-1">
          <div class="sticky top-0 flex items-center justify-between px-2 pt-2 sm:top-0 bg-neutral-900">
            <h3 class="px-2 py-1 text-sm font-bold tracking-tight text-neutral-400">
              Lists
            </h3>
            <a href="/lists/create" class="block px-3 text-xl rounded hover:bg-neutral-800 text-neutral-400">+</a>
          </div>
          <ul class="pb-2">
            <li>
              <a class="block px-4 py-1 overflow-hidden text-ellipsis whitespace-nowrap hover:bg-neutral-800" href="/bookmarks/unsorted">Unsorted bookmarks</a>
            </li>
            <li>
              <a class="block px-4 py-1 overflow-hidden text-ellipsis whitespace-nowrap hover:bg-neutral-800 text-neutral-400" href="/lists/unpinned">Unpinned lists</a>
            </li>
          </ul>
        </div>
        <header class="sticky bottom-0 flex justify-between p-2 leading-8 bg-neutral-900">
          <a href="/" class="px-2 font-bold rounded hover:bg-neutral-800">testuser</a>
          <form action="/logout" method="post">
            <button class="rounded px-3 text-neutral-400 hover:bg-neutral-800">
              Logout
            </button>
          </form>
        </header>
      </aside>
    </div>
  </body>
